### DISCUSSION TITLE
NATS fuzzing tests and output

### DISCUSSION THREAD

**Discussion by AlexeiSosnov** [nats-io/nats-server#6201] [Q&A]
Hello,

Can anybody please tell, are there any known community fuzzing tests of NATS? I did not find any fuzzing tests (testing.F) in NATS codebase itself. When NATS used in some perimeters, fuzzing could be very desirable. 
Are there also any plans to add  fuzzing to NATS codebase itself?

**Comment by leeriorio** [comment]
Hello!
I am also looking for an answer to this question and I hope you are still interested in this question. In official repo there is at least one fuzz test. Added in 2020 with [f7cbac7](https://github.com/nats-io/nats-server/commit/f7cbac740257b5c0e3888451f5bb0f21a968db78)

Moreover, test collection became bigger with [#6728](https://github.com/nats-io/nats-server/pull/6728) with some more fuzz tests by @astmix

**Reply by astmix** [reply]
I adapted fuzz.go into parser_fuzz_test.go. I think fuzz.go is no longer needed, and we can simply delete this file
